docs/core/overview.mdx

```diff
@@ -0,0 +1,299 @@
+---
+title: "Core Concepts Overview"
+description: "Understanding Suga's architecture and core concepts"
+---
+
+Suga is built on a clear separation of concerns that enables both developer productivity and platform governance. Understanding these core concepts will help you make the most of the platform.
+
+## The Suga Architecture
+
+Suga operates through three key layers that work together to transform your application designs into deployed infrastructure:
+
+```mermaid
+graph TB
+    A[Projects<br/>What you want] --> D[Platforms<br/>How to build it]
+    D --> B[Plugins<br/>Cloud resources]
+    B --> C[Terraform<br/>Deployed infrastructure]
+
+    style A fill:#2BC65F,stroke:#1E9A47,stroke-width:2px,color:#fff,font-weight:500
+    style D fill:#25B355,stroke:#1E9A47,stroke-width:2px,color:#fff,font-weight:500
+    style B fill:#1E9A47,stroke:#178A3D,stroke-width:2px,color:#fff,font-weight:500
+    style C fill:#178A3D,stroke:#125E2F,stroke-width:2px,color:#fff,font-weight:500
+```
+
+### 1. Projects: What You Want
+
+[Projects](/core/projects) are your application specifications - declarative YAML files that describe what your application needs:
+
+- **Services** - Application containers that run your code
+- **Buckets** - Cloud storage for files and assets
+- **Databases** - PostgreSQL databases for structured data
+- **Entrypoints** - HTTP routing and CDN configuration
+
+Projects are **deployment-agnostic** - they describe requirements without specifying how they're implemented.
+
+```yaml title="Example: A simple web application"
+target: suga/aws@1
+name: my-app
+
+services:
+  api:
+    dev:
+      script: npm run dev
+
+buckets:
+  uploads:
+    access:
+      api: [read, write]
+
+entrypoints:
+  web:
+    routes:
+      /api/: 
+        name: api
+```
```

Comment:
Possibly including a diagram demonstrating how the generated clients and suga server wrap application code in deployment. Having the `target` be included here next to the words `deployment agnostic` does look a little odd.

Reply:
Removed the target for now and created a new issue to have another go at this section in the next update